extensions/prototypes/labgraph_monitor/src/components/Graph/Graph.tsx

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useEffect, useState } from 'react';
+import ReactFlow, {
+    ReactFlowProvider,
+    Controls,
+    Background,
+    Edge,
+    Node,
+    isNode,
+    isEdge,
+} from 'react-flow-renderer';
+import { TGraphElement } from './types/TGraphElement';
+import { useUIContext } from '../../contexts';
+import { layoutGraph } from './util/layoutGraph';
+import { RootState } from '../../redux/store';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+    setPanel,
+    setTabIndex,
+    setSelectedNode,
+    setSelectedEdge,
+} from '../../redux/reducers/config/configReducer';
+import WS_STATE from '../../redux/reducers/ws/enums/WS_STATE';
+import { Box } from '@mui/material';
+
+/**
+ * A component that represents LabGraph computational graph.
+ *
+ * @returns {JSX.Element}
+ */
+const Graph: React.FC = (): JSX.Element => {
+    const { layout } = useUIContext();
+    const { mockGraph } = useSelector((state: RootState) => state.mock);
+
+    const { connection, graph: realtimeGraph } = useSelector(
+        (state: RootState) => state.ws
+    );
+
+    const serializedGraph =
+        connection === WS_STATE.CONNECTED ? realtimeGraph : mockGraph;
+
+    const [graph, setGraph] = useState<Array<TGraphElement>>(
+        [] as Array<TGraphElement>
+    );
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const { nodes } = serializedGraph;
+        const deserializedGraph = [] as Array<TGraphElement>;
+        if (!nodes) return;
+        for (const [name, data] of Object.entries(nodes)) {
+            deserializedGraph.push({
+                id: name,
+                data: {
+                    label: name,
+                },
+                position: { x: 0, y: 0 },
+                targetPosition: layout === 'horizontal' ? 'left' : 'top',
+                sourcePosition: layout === 'horizontal' ? 'right' : 'bottom',
+                style: {
+                    display: 'flex',
+                    justifyContent: 'center',
+                    alignItems: 'center',
+                    width: '150px',
+                    height: '150px',
+                    borderRadius: '50%',
+                    fontWeight: '500',
+                    fill: 'currentColor',
+                },
```

Comment:
nitpick might be best to extract this outside the useEffect logic, since the style doesn't depend on any dynamic values

Reply:
Done